### ISSUE TITLE
Embedded Overlay Processing fails with Index out of Range

### ISSUE THREAD

**Issue by Ed55** [fo-dicom/fo-dicom#138]
I am no pixel guy but I accidentially I clicked on Dicom Dump Tools/View on the image file I uploaded.
(renamed to .png)
![Uploading fdEmbeddedOverlayIssue.png…]()

The image seems to be ok, other viewers display it correctly.
Some calculations seem to go wrong in DicomOverlayData IByteBuffer Load()  and may be later as well.
In most of my use-cases I do not care about pixel, but for the sake of fo-dicom I just want to report it.
 
Uups, seems the upload did not worked.

**Comment by Ed55** [comment]
Embedded overlays are retired since 2005, but there are old systems as well as old images still out there ...
It might be related to #110 , but the uploaded images all have 60xx,0102 Overlay Bit Position value 0 which is a DICOM violation whithout 60xx,3000 or without having something strange as Bits Allocated 16, Bits Stored 12 and High Bit 15.

I try again to upload 
![fdembeddedoverlayissue](https://cloud.githubusercontent.com/assets/15705674/11079372/282f3cdc-880c-11e5-8dc9-0a3ff4166d74.jpg) rename to zip and unpack.

This one should display  "LT" in the upper right corner:

![overlaydisplayed](https://cloud.githubusercontent.com/assets/15705674/11079443/b6a194ec-880c-11e5-8a45-467a0d63cf84.PNG)

of course in magenta :-)

**Comment by anders9ustafsson** [comment]
Thanks for reporting, @Ed55 . So the image display crashes, or? (Haven't check myself yet.)

**Comment by Ed55** [comment]
No, the exception is caught and displayed.
I have tried to dig into it but must confirm that I am definitely no pixel guy.
The exception is thrown in DicomOverlayData on iByteBuffer Load() 
else if (pixels.BitsAllocated == 16)
{
...
if ((data[n] & mask) != 0) bits[i] = true;      where n exceeds the array range

**Comment by Ed55** [comment]
I do not want to dig into the complete display pipeline.
But these are my findings:
The line "output.Add(new DicomTag(overlay.Group, DicomTag.OverlayBitPosition.Element), (ushort)0);"
in DicomTranscoder.ProcessOverlays overwrites the OverlayBitPostion to 0 so that (at least in this case) the DicomOverlayData.BitPosition returns 0 as well so that the the var mask in DicomOverlayData.Load() is 0 as well.
I removed this line for this test.

Next: the iterations in  DicomOverlayData.Load()  and pixels.BitsAllocated == 16 are somehow miscalculated.
Ignoring OverlayOrigin (which is 1/1 in this case anyway) I have tried to make it simple first and replaced in DicomOverlayData.Load() the complete loop
for (int y = oy; y < oh; y++)
{
...
}
with
                   int ii = 0;
                    foreach (var item in data)
                    {
                        if ((item & mask) != 0)
                        {
                            bits[ii] = true;
                        }
                        ii++;
                    }
That's of course no final solution, but it may help

![quickfixoverlayresult](https://cloud.githubusercontent.com/assets/15705674/11088596/dc5e8454-8863-11e5-8ca4-c62f2803c1da.PNG)
